docs/dg/dev/troubleshooting/troubleshooting-performance-issues/redis-session-lock-excpetions.md

```diff
@@ -1,18 +1,36 @@
 ---
-title: Redis session lock exceptions
-description: Redis Session Lock Exceptions
+title: Redis session lock
+description: Redis Session Lock
 template: troubleshooting-guide-template
 redirect_from:
 - /docs/scos/dev/troubleshooting/troubleshooting-performance-issues/redis-session-lock.html
-last_updated: Aug 1, 2024
+last_updated: Sep 28, 2024
 ---
 
-It's impossible to mitigate `LockCouldNotBeAcquiredExecption` exceptions on the infrastructure level, for example, by increasing CPU or memory resources. To make sure an application reaches expected scaling targets, during performance and load testing, take into account the scaling challenges described in this document.
+A session lock is needed to prevent multiple processes from accessing or modifying session data simultaneously, ensuring data integrity and consistency. 
+It helps avoid race conditions and data corruption in high-concurrency environments.
+While a session is being accessed by a process, other processes can't access it; the session is *locked*. 
+If a process can't lock the session, it tries to acquire it by using a *spinlock* strategy. 
+It waits for 0.01 seconds by default and then retries acquiring a lock to the session. 
+These attempts continue until the timeout, which is 0.8 from `max_execution_time`. 
+
+The handler is configured with:
+```php
+// config/Shared/config_default.php
+$config[SessionConstants::YVES_SESSION_SAVE_HANDLER] = SessionRedisConfig::SESSION_HANDLER_REDIS_LOCKING;
+```
 
+## Probable causes
+Use cases:
+- Users clicking fast, especially in the app that allows activating multiple parallel requests;
+- Requests being interrupted (but lock stays) because of aborting the connection (network or manual intentional and unintentional act, aka clicking “Esc“ when page is loading);
```

Comment:
Since now we know more, to not go too far into tech details, let's maybe say:  
- Long-running (or slow performing) requests that have to deal with the session

Reply:
added